sdcm/cluster_gce.py

```diff
@@ -341,42 +341,59 @@ def _create_instance(self, node_index, dc_idx, spot=False):
                                                "ssh-keys": f"{username}:ssh-rsa {public_key}", },
                                   ex_service_accounts=self._service_accounts,
                                   ex_preemptible=spot)
-        try:
-            instance = self._gce_services[dc_idx].create_node(**create_node_params)
-        except GoogleBaseError as details:
-            if not spot:
-                raise
-            self.log.warning('Unable to create a spot instance, try to create an on-demand one: %s', details)
-
-            # May happen that spot instance was preempted during instance creating. This instance won't be
-            # destroyed but stopped and stays with the name in the instances pool. In this case we should destroy this
-            # instance before creating new one
-            if "Instance failed to start due to preemption" in str(details) and \
-                    (failed_instance := self._get_instances_by_name(dc_idx=dc_idx, name=name)):
-                try:
-                    self._gce_services[dc_idx].destroy_node(node=failed_instance, destroy_boot_disk=True)
-                # if the instance is destroyed already - ignore the exception
-                # otherwise - raise exception
-                except ResourceNotFoundError as exc:
-                    if 'notFound' not in str(exc):
-                        raise
 
-            create_node_params['ex_preemptible'] = spot = False
-            instance = self._gce_services[dc_idx].create_node(**create_node_params)
+        instance = self._create_node_with_retries(name=name,
+                                                  dc_idx=dc_idx,
+                                                  create_node_params=create_node_params,
+                                                  spot=spot)
 
         self.log.info('Created %s instance %s', 'spot' if spot else 'on-demand', instance)
         if gce_job_default_timeout:
             self.log.info('Restore default job timeout %s' % gce_job_default_timeout)
             self._gce_services[dc_idx].connection.timeout = gce_job_default_timeout
         return instance
 
+    @retrying(n=3,
+              sleep_time=900,
+              allowed_exceptions=(GoogleBaseError, ResourceNotFoundError),
+              message="Retrying to create a GCE node...",
+              raise_on_exceeded=True)
+    def _create_node_with_retries(self,
+                                  name: str,
+                                  dc_idx: int,
+                                  create_node_params: dict[str, Any],
+                                  spot: bool = False) -> GCENode:
+        try:
+            instance = self._gce_services[dc_idx].create_node(**create_node_params)
+            return instance
+        except GoogleBaseError as gbe:
+            if not spot:
+                raise gbe
```

Comment:
It can be just `raise` to reraise a caught exception.

Reply:
Fixed